docs/DeveloperGuide.md

```diff
@@ -4,30 +4,136 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+Ideas and structure for the Developer Guide: https://se-education.org/addressbook-level3/DeveloperGuide.html
+
+Ideas and structure for the User Guide: https://se-education.org/addressbook-level3/UserGuide.html
+
+Additional Packages used: JSON
+
+## Setting up and getting started
 
+
+## Design & implementation
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Add Medication Command
+The add medication command extends from the Command parent class and implements the following operations:
+- execute(MedicationManager, DailyMedicationManager, Ui) - Adds the medication object into the respective medication managers.
+- setMedicineAttributes() - Sets the medicine object's attributes to be added to the medicine managers. 
+- assertionTest(MedicationManager, DailyMedicationManager) - Asserts that medicine has been added to both medication managers.
+- parseStringToValues(Arguments) - Parses string input for medicine quantity and dosage into double type.
+
+Given below is an example usage scenario and how the add command behaves at each step.
+- Step 1. To be added.
+- Step 2. To be added.
+
+---
+
+### List Medication Command
+The list medication command extends from Command parent class and contains the following methods:
+- execute(MedicationManager medicationManager) - Executes the list command and performs its specific task, -t.
+```Where the task can be either 'list -t all' to list all medications or 'list -t today' to list medications for the day, which is divided into three categories -> Morning, Afternoon and Evening. ```
+
+The 'list -t' command requires the following:
+1. 'all' - to run printAllMedications() from the MedicationManager.
+2. 'today' - to run printMedications() from the DailyMedicationManager.
+3. 'today -m' to run printTodayMedications(List<Medication> medications, List<DailyMedication> subList, String period)
+   from the DailyMedicationManager
+4. 'today -a' to run printTodayMedications(List<Medication> medications, List<DailyMedication> subList, String period)
+   from the DailyMedicationManager
+5. 'today -e' to run printTodayMedications(List<Medication> medications, List<DailyMedication> subList, String period)
+   from the DailyMedicationManager
```

Comment:
Perhaps clarify what is the difference between inputting "today -m", "today -a" and "today -e"?